spec/predicates/human-review-vcs.md

```diff
@@ -0,0 +1,138 @@
+issuing the review.
+
+## Schema
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [
+        {
+            "name": "<REPO>",
+            "digest": { "gitCommit": "<HASH>" },
+            "uri": "<LINK TO REPO>"
+        },
+    ],
+    "predicateType": "http://in-toto.io/attestation/human-review/vcs/v0.1",
+    "predicate": {
+        "reviewers": ["<ResourceDescriptor>", ...],
+        "target": "<ResourceDescriptor>",
+        "reviewLink": "<LINK TO REVIEW>",
+        "reviewTime": "<TIMESTAMP>",
+        "annotations": {...}
+    }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the
+[in-toto Attestation Framework's parsing rules](../v1/README.md#parsing-rules).
+
+### Fields
+
+`reviewers` _array of ResourceDescriptor_, _required_
+
+Indicates uniquely the identity of one or more reviewers. MUST NOT be used over
+the attestation's signature to validate the source of the attestation. This
+field enables scenarios where the attestation is issued and signed by the VCS
+review system. Each entry must include immutable information that is always
+mapped to the reviewer. The digest may be skipped in the ResourceDescriptor,
+using the name and URI fields to record the username and immutable ID.
+
+`target` _ResourceDescriptor_, _required_
+
+Records the target location for the proposed changeset. In Git repositories, the
+field must contain the base Git reference (i.e., the base branch) AND the
+current commit ID at the tip of that branch.
+
+`reviewLink` _URI_, _optional_
+
+Contains a link to the full review. Useful to point to information that cannot
+be captured in the attestation.
+
+`reviewTime` _Timestamp_, _optional_
+
+Indicates the time of _last review_ creation. If the attestation is for a single
+review, this is equivalent to when that review was created. With multiple
+reviewers, the attestation is re-created for each new review and the time
+reflects that of the latest review.
+
+`annotations` _object_, _optional_
+
+Optional, opaque field to add annotations to the review. This object can be used
+to record review system specific fields.
+
+## Example
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [{
+        "name": "in-toto-python",
+        "digest": {
+            "gitCommit": "b0bd8ecab5607e174fa403e002c74a666e7edd51"
+        },
+        "uri": "https://github.com/in-toto/in-toto"
+    }],
+    "predicateType": "http://in-toto.io/attestation/human-review/vcs/v0.1",
+    "predicate": {
+        "reviewers": [{
+            "name": "lukpueh",
+            "uri": "https://api.github.com/user/589324"
+        }],
+        "target": {
+            "name": "refs/heads/develop",
+            "digest": {
+                "gitCommit": "330500b54433de4f6f9575676b67738b98ba5e54",
+            },
+        },
+        "reviewLink": "https://github.com/in-toto/in-toto/pull/503#pullrequestreview-1341209941",
```

Comment:
This spec doesn't describe the actual change set that was reviewed, and therefore places a lot of the load on this `reviewLink`.

For a given PR, the diffs reviewed and voted upon by each reviewer regularly change (unless rules are enforced to dismiss "stale" reviews).

Mechanically this review link is reverencing the critical "provenance" data maintained by the PR product (GitHub in this case). 
Pull request data is saved as evidence justifying the existence of the new repo version.

PR data is not quite as nice as signed attestation statements like this though.  
PR data can change over time, either by-design (EG: by rerunning checks on the same refs) or maliciously if an attacker gains access to the DB.

Reply:
The idea is to infer the diff from the subject (which identifies the tip of the reviewed topic branch) and the base recorded in the predicate. I leaned away from including the full diff inline due to length considerations. `reviewLink` is included ATM for completeness, any information that isn't considered vital to the review process for the changeset but may still be necessary.

You make a good point about the iterative nature of reviews and that's where a tighter alignment with GitHub mechanics would be helpful to answer questions like when the attestation is signed and generated. If an attestation is generated at merge time, for example, it captures the final states which is _probably_ what we care about?

Reply:
TODO: `target` needs to be expanded to handle current tip of the base branch and the merge base for diff inference.